### ISSUE TITLE
UI: annotations for implementations of interfaces

### ISSUE THREAD

**Issue by BIKA-C** [golang/vscode-go#2299]
Since go does not have an explicit declaration of which interfaces one implements, it is hard to follow someone's code. I know that there is a command to show all the interfaces, but it is not very intuitive nor efficient.

Goland has the feature of showing a gutter when a function implements an interface. It would be nice to show maybe a codelens over the implementation and the type, or a similar solution as Goland.

Codelens:
![pasted](https://user-images.githubusercontent.com/50717946/174982252-3d672678-052b-4aea-b4f2-e6458d10bb1c.png)

Goland:
<img width="744" alt="image" src="https://user-images.githubusercontent.com/50717946/174978822-cecb02b8-ba0c-4c8f-8f87-3da61ac66eca.png">

I found an extension that does kinda the opposite which is not very helpful in most scenario: [galkowskit/go-interface-annotations](https://github.com/galkowskit/go-interface-annotations)
<img width="718" alt="image" src="https://user-images.githubusercontent.com/50717946/174983652-4ad9ef84-13ab-4401-8862-e51f6dd0b1a0.png">

**Comment by 2opremio** [comment]
That's exactly how `rust-analyzer` handles traits

**Comment by icearith** [comment]
this feature seems exactly what I want.

**Comment by Shawn-Huang-Tron** [comment]
This feature is very useful and we use it frequently. A lot of my friends doesn't like vscode-go just because of the leak of this feature.So they change to goland.Hope to add such a feature.

**Comment by Tony-Sol** [comment]
+1 
this would be an extremely usefull

**Comment by hyangah** [comment]
cc @golang/tools-team This can be expensive depending on the size of the search space, but given that there are already similar functionalities (Find All Implementations, or LSP `textDocument/implementation`), I guess gopls has sufficient information to package this info as a codelens. Is there any concern?

**Comment by findleyr** [comment]
We have this in our long-term roadmap for gopls (along with a references codelens), most likely once we build out indexes that can satisfy these queries cheaply. Moving to the gopls issue tracker.
